x-pack/plugins/apm/public/components/app/Settings/anomaly_detection/jobs_list.tsx

```diff
@@ -121,12 +117,17 @@ export const JobsList = ({
       <EuiSpacer size="l" />
       <ManagedTable
         noItemsMessage={
+          // loading state
           isLoading ? (
             <LoadingStatePrompt />
-          ) : hasFetchFailure ? (
-            <FailureStatePrompt />
+          ) : // Handled error
+          errorCode ? (
+            MLErrorMessages[errorCode]
+          ) : // Unhandled error
+          hasFetchFailure ? (
+            unhandledErrorText
           ) : (
-            <EmptyStatePrompt />
+            emptyStateText
```

Comment:
I think it's a bit difficult to read this kind of ternary

Reply:
I agree. Will extract to a function